python/desc/imsim/trim.py

```diff
@@ -99,14 +103,24 @@ def get_object_entries(self, chip_name, radius=0.18, sort_magnorm=True):
                     number of sersic objects for minsource application)
 
         """
+        self.trimmer.logger.debug("computing object offsets from %s center",
+                                  chip_name)
         ra0, dec0 = self.compute_chip_center(chip_name)
         seps = degrees_separation(ra0, dec0, self._ra, self._dec)
-        index = np.where(seps < radius)
+        if chip_name in self.trimmer.drawn_objects:
+            not_drawn = [not x in self.trimmer.drawn_objects[chip_name]
+                         for x in self._ids]
+            index = np.where((seps < radius) & not_drawn)
+            self.trimmer.logger.debug("avoiding drawn objects")
+            self.trimmer.logger.debug(index)
+        else:
+            index = np.where(seps < radius)
 
         # Collect the selected objects.
         selected = [self.object_lines[i] for i in index[0]]
         if sort_magnorm:
             # Sort by magnorm.
+            self.trimmer.logger.debug('sorting by magnorm')
```

Comment:
Is this going to happen for all runs?  Only ones when multiprocessing is used?

It looks like InstCatTrimmer is always being called from parsePhoSimInstanceFile.

For simple debugging for individuals (rather than big production) having the processing order different than what is in the file will be confusing and will make debugging difficult.  If it is always going to happen maybe we could make the default to be non-sorted and then allow a flag?

Reply:
Yes, but maybe the last time I did it and really cared about the order was before the sorting started?  If you have a catalog and you want to go through and track what it is doing line by line it  would be confusing if the order wasn't what you input.

Also, when doing testing I often will do (say) the 1st 2000 objects for speed. What happens in that case? Do you get the 2000 brightest objects? This might actually explain some confusing things I have seen recently when I predicted times based on running a fraction of a file.

True about things changing if we break things up.  Maybe we should set up three modes if we do that: "as read", "magnum sorted", and "interleaved"?

Reply:
I added the config parameter.  We can revisit this when we look into breaking up the bright objects.